src/utils/copy/createCogAbilityOptions.ts

```diff
@@ -406,6 +455,12 @@ export const cogAbilityOptions: CogAbilitySection[] = [
             effect:
               "This Cog is cannot be knocked prone. When this Cog is forcibly moved, halve the distance of the effect (this ability stacks with similar effects).",
           },
+          {
+            name: "Thorns",
+            cost: "X",
+            effect:
+              "Whenever this Cog is targeted by a melee attack from an adjacent attacker, the attacker takes an unavoidable {{X}} damage (up to {{L}}) before resolving the attack.",
```

Comment:
When you have `X` in the effect string, you should use square brackets for the formula, since they will be replaced with the values the user enters in the text box.